src/message/messagesActions.js

```diff
@@ -24,7 +25,10 @@ export const doNarrow = (narrow: Narrow, anchor: number = FIRST_UNREAD_ANCHOR) =
   dispatch(navigateToChat(narrow));
 
   const allNarrows = getAllNarrows(state);
-  const messagesForNarrow = allNarrows[JSON.stringify(narrow)] || NULL_ARRAY;
+  const messages = getMessages(state);
+  const messagesForNarrow = (allNarrows[JSON.stringify(narrow)] || NULL_ARRAY).map(
+    id => messages[id],
+  );
```

Comment:
Orthogonal to this change, but reading this code in the course of rereading this whole change I notice that these lines would be simpler as `getFetchedMessagesForNarrow(narrow)(state)`.